File: app/src/main/java/wangfeixixi/github/com/wiget/MainActivity.java
package wangfeixixi.github.com.wiget;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Toast;

import wangfeixixi.github.com.bottomtimedialog.BottomDialogListener;
import wangfeixixi.github.com.bottomtimedialog.BottomTimeDialog;
import wangfeixixi.github.com.bottomtimedialog.BottomTimeShowDialog;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main_activity);


        findViewById(R.id.btn_timepicker).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                pop();
            }
        });
    }

    public void pop() {
        BottomTimeDialog bottomTimeDialog = new BottomTimeDialog(MainActivity.this);
        bottomTimeDialog.setOnItemClickListener(new BottomDialogListener() {
            @Override
            public void onItemClick(String text) {
                Toast.makeText(MainActivity.this, text, Toast.LENGTH_SHORT).show();
            }
        });
        bottomTimeDialog.show();
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.btn_timepicker:
                startActivity(new Intent(this, TimePikerActivity.class));
                break;
            case R.id.btn_editview:
                startActivity(new Intent(this, EditViewActivity.class));
                break;
            case R.id.btn_calendar:
                startActivity(new Intent(this, CalendarActivity.class));
                break;
            case R.id.btn_sweet_dialog:
                startActivity(new Intent(this, SweetDialogActivity.class));


                break;
        }
    }

}

File: app/src/main/java/wangfeixixi/github/com/wiget/MainActivity.kt
package wangfeixixi.github.com.wiget

import android.graphics.Color
import android.os.Bundle
import android.os.Handler
import android.os.Message
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.Toast
import kotlinx.android.synthetic.main.main_activity.*
import wangfeixixi.share.bottomtimedialog.BottomTimeDialog
import java.util.*

class MainActivity : AppCompatActivity() {

    private val iamgeUrl = "http://p3tnh1dg6.bkt.clouddn.com/ca1349540923dd548b8a28fbda09b3de9d8248ac.jpg"

    var mPower: Int = 0
    val mHandler = object : Handler() {
        override fun handleMessage(msg: Message) {
            super.handleMessage(msg)
            when (msg.what) {
                0 -> {
                    mPower = mPower + 5
                    horizontalBattery.power = (mPower)
                    if (mPower == 50) {
                        mPower = 0
                    }
                }
                else -> {
                }
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main_activity)

        verticalBattery!!.setColor(Color.GRAY)
        verticalBattery!!.setPower(90)

        Timer().schedule(object : TimerTask() {
            override fun run() {
                mHandler.sendEmptyMessage(0)
            }
        }, 0, 100)
    }

    fun pop() {
        val bottomTimeDialog = BottomTimeDialog(this@MainActivity)
        bottomTimeDialog.setOnItemClickListener { text -> Toast.makeText(this@MainActivity, text, Toast.LENGTH_SHORT).show() }
        bottomTimeDialog.show()
    }


}
